src/compute-types/src/plan/mod.rs

```diff
@@ -95,6 +95,9 @@ pub struct AvailableCollections {
     /// column permutation mapping
     #[proptest(strategy = "prop::collection::vec(any_arranged_thin(), 0..3)")]
     pub arranged: Vec<(Vec<MirScalarExpr>, BTreeMap<usize, usize>, Vec<usize>)>,
+    /// The types of the columns in the raw form of the collection,
+    /// if known, or empty, otherwise.
+    pub types: Vec<ColumnType>,
```

Comment:
Fixed now! Documenting what we discussed offline: We really want to use `Option<Vec<_>>` at this point, since we support relations with no columns in Materialize.